### DISCUSSION TITLE
ERL_EPMD_PORT in rabbitmq-env.bat ignored for Windows Service

### DISCUSSION THREAD

**Discussion by danstur** [rabbitmq/rabbitmq-server#11080] [Questions] [answered]
### Describe the bug

Using `SET ERL_EPMD_PORT=...` in the `rabbitmq-env.bat` file does not seem to have any effect when running RabbitMQ as a service under Windows. The file is correctly read (i.e. introducing a syntax error leads to errors) and using `rabbitmq-server.bat` to run the program interactively honors the variable as expected.

This might be a similar bug to #1709  just on the Windows side of things.

### Reproduction steps

1. Ensure that epmd is not already running (`Get-Process -Name epmd | Stop-process`)
1. Add `SET ERL_EPMD_PORT=50000` or any other non-standard port in rabbitmq-env.bat
1. Run  .\rabbitmq-service.bat install;  .\rabbitmq-service.bat start

### Expected behavior

epmd should be running under port 50000 instead of 4369.

### Additional context

RabbitMQ 3.1.13
Erlang 26.2.4
Windows 10 x64 22H2 (19045.4291)

**Comment by lukebakken** [comment, marked as answer]
Hello, thanks for using RabbitMQ.

You will have to set `ERL_EPMD_PORT` as a system-wide environment variable, then re-install RabbitMQ.

When running as a service, the [`rabbitmq-env-conf.bat`](https://www.rabbitmq.com/docs/configure#rabbitmq-env-file-windows) file is not used.

Let me know how that works in your environment.

**Reply by danstur** [reply]
Thanks for the quick reply. Too bad, that would simplify a few things for us.

It'd be helpful if the documentation could be made a bit clearer on this point though. Currently it says:

> Alternatively it is possible to use a file named rabbitmq-env-conf.bat to define environment variables that will be used by the broker. Its [location](https://www.rabbitmq.com/docs/configure#config-location) is configurable using the RABBITMQ_CONF_ENV_FILE environment variable.
> 
> Windows service users will need to re-install the service if configuration file location or any values in ``rabbitmq-env-conf.bat` changed. Environment variables used by the service would not be updated otherwise.

[Source](https://www.rabbitmq.com/docs/configure#rabbitmq-env-file-windows) 

That was a bit confusing to me on first glance, but looking at the batch files it makes sense - the file is only used for initial configuration and only the few values passed to erlsrv will be persisted.  That's why setting NODENAME in it was having an effect (and why introducing a syntax error caused problems for the installation where it _is_ read), but most other variables will be ignored.

**Reply by lukebakken** [reply]
Thank you for the follow-up. If you have time, please feel free to open a PR to improve the documentation. I created the following issue to track this - https://github.com/rabbitmq/rabbitmq-website/issues/1895

It would be great if we had more time to focus on Windows but it's just not as popular with RabbitMQ users who pay for support (and thus pay for development).

**Reply by danstur** [reply]
I'll see if I can come up with something that makes it a bit clearer.
